solr/core/src/test/org/apache/solr/update/UpdateLogTest.java

```diff
@@ -218,6 +223,23 @@ public void testApplyPartialUpdatesWithDelete() throws Exception {
     }
   }
 
+  @Test
+  public void testLogCleanupWhenNewTransactionLogConstructionFails() throws Exception {
+    // Simulate the appearance of a new log file after UpdateLog.init().
+    Path tlogDir = Path.of(ulog.getTlogDir());
+    long lastLogId = scanLastLogId(tlogDir);
+    String newLogName =
+        String.format(
+            Locale.ROOT, UpdateLog.LOG_FILENAME_PATTERN, UpdateLog.TLOG_NAME, lastLogId + 1);
+    Path newLogPath = tlogDir.resolve(newLogName);
+    createLogFile(newLogPath);
+
+    // Add a doc and expect no "New transaction log already exists" error when creating the new
+    // transaction log.
+    ulogAdd(
```

Comment:
Without the fix, this call throws the "New transaction log already exists" exception.